docs/content/guides/developer/getting-started/data-serving.mdx

```diff
@@ -0,0 +1,135 @@
+
+### gRPC API
+
+As mentioned above, gRPC API will replace the JSON-RPC on Full nodes, such that JSON-RPC will be deprecated when gRPC API is generally available. Apart from the message & request format changes between the two, the gRPC API will come with a couple of key functional differences:
+
+- It will have streaming or subscription API endpoints such that you could consume real-time streaming data in your application without having to poll for those records. This support will be a proper replacement of the deprecated Websockets support in JSON-RPC.
+- It will not have implicit fallback on the above mentioned [Sui Foundation managed key-value store for historical transaction data](/guides/operator/data-management.mdx#sui-full-node-key-value-store-backup). Full node operators, RPC providers, and Data indexer operators will be encouraged to run their own instance of a similar archival store, which you would be able to explicitly define a dependency on to get the relevant historical data.
+
+:::info
+
+The gRPC API is in alpha and not recommended for production use. Its public documentation is coming soon. Until then, you can refer to [this technical proto spec](https://github.com/MystenLabs/sui/blob/main/crates/sui-rpc-api/proto/sui.node.v2.proto).
+
+:::
+
+**High-level timeline**
+
+The target times indicated below are tentative and subject to updates based on project progress and your feedback.
+
+| **Tentative time** | **Milestone** | **Description** |
+| -------- | ------- | ------- |
+| March 2025 | Beta release of initial set of polling-based APIs | You can start validating the initial gRPC integration from your application and share feedback on the improvements you would like to see. |
+| May 2025 | Beta release of streaming APIs and the remaining set of polling-based APIs | If your use case requires streaming low-latency data, this would be apt time to start validating that integration. Also, the API coverage will be functionality wise complete at this point, so you can start migrating your application in the non-production environments. |
+| July 2025 | GA release of polling-based APIs with appropriate SDK support | You can start migration and cutover of your application in the production environment, with the support for streaming APIs still in beta. |
+| September 2025 | GA release of scalable streaming APIs | If your use case requires streaming low-latency data, you can now use those APIs at scale. **JSON-RPC will be deprecated at this point and migration notice period will start.** |
+| June 2026 | End of migration timeline | **JSON-RPC will be fully deactivated at this point.** This timeline assumes about 9 months of migration notice period. |
+
+See [below](#when-to-use-grpc-vs-graphql-with-indexer-20) for a comparison with GraphQL RPC with Indexer 2.0.
+
+### GraphQL RPC with Indexer 2.0
```

Comment:
Should "Indexer 2.0" be its own section? People may also be curious about when they should use that vs the indexing framework, vs GraphQL.

Reply:
Carved it out into its own section, but kept the release timeline consolidated for both.